File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        


        maxcur  = maxfar  = 0 
        for i in range(1,len(prices)):
            maxcur = max( 0 , maxcur + prices[i]-prices[i-1] )
            maxfar = max( maxcur , maxfar)
        return maxfar 

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int maxcur =0 ;
        int maxfar = 0;
        
        for(int i=1 ; i<prices.size() ; ++i){
            
            maxcur = max( 0 , maxcur + prices[i]- prices[i-1]);
            maxfar = max( maxfar , maxcur );
        }
        return maxfar ;
        
        
        
    }
};